bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -347,6 +366,55 @@ public boolean getAndResetHeartbeat() {
         return result;
     }
 
+    public void reconnectWebsocket() {
+        if (webSocketReconnectionPollingJob == null) {
+            webSocketReconnectionPollingJob = scheduler.scheduleWithFixedDelay(this::reconnectWebsocketJob, 0, 30,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    public void reconnectWebsocketJob() {
+        switch (yioRemoteDockActualStatus) {
+            case COMMUNICATION_ERROR:
+                logger.debug("Reconnecting YIORemoteHandler");
+                try {
+                    disposeWebsocketPollingJob();
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
```

Comment:
Why `CONFIGURATION_PENDING` here? This is not valid for an `OFFLINE` status.

Reply:
Which status is than better for reconnection?

Reply:
If "reconnection" means that the communication was lost, it should be `OFFLINE.COMMUNICATION_ERRROR`.

Reply:
changed